src/waves/NcurseOneLineRenderer.cpp

```diff
@@ -0,0 +1,146 @@
+#include "NcurseOneLineRenderer.hpp"
+#include "raii.h"
+#include <algorithm>
+#include <iostream>
+#include <cassert>
+#include <ncurses.h>
+
+namespace {
+
+auto create_pad(int height, int width)
+{
+    return make_raii(
+        [height, width] {
+            auto rslt = newpad(height, width);
+            if (!rslt)
+                throw std::runtime_error{"Ncurses error: unable to initialize pad"};
+            return rslt;
+        }
+      , [] (auto pad) { delwin(pad); }
+    );
+}
+
+auto create_screen()
+{
+    return make_raii(
+        []{
```

Comment:
```suggestion
        []
        {
```

Reply:
I got you point; But I'm pretty flexible when it relates to lambdas formatting (a syntax in general can lead to pretty strange constructions). In this particular case I can't see any good reason to follow regular functions formatting style rules. The body is too short as well as a capture list. So I would prefer to avoid excessive verbosity.

So, stylish discussions can be very intensive and interesting, but usually they aren't quite productive. I'm resolving the topic to let the mr to be merged.